api/jobs/handlers.py

```diff
@@ -67,12 +67,59 @@ def get_invocation(self, _id):
 
     @require_login
     def suggest(self, _id, cont_name, cid):
-        cr = ContainerReference(cont_name, cid)
-        if not self.superuser_request:
-            cr.check_access(self.uid, 'ro')
+        """
+        Given a container reference, return display information about parents, children and files
+        as well as information about which best match each gear input
+
+        Container types acceptable for reference:
+          - Groups
+          - Projects
+          - Subjects
+          - Sessions
+          - Acquisitions
+          - Analyses
+
+        NOTE: Access via this endpoint is not logged. Only information necessary for display should be returned.
+        NOTE: Subject level is supported ahead of official separation in DB and API routes.
+        """
 
+        # Do all actions that could result in a 404 first
         gear = get_gear(_id)
-        return suggest_container(gear, cont_name+'s', cid)
+        if not gear:
+            raise APINotFoundException('Gear with id {} not found.'.format(_id))
+
+        storage = cs_factory(cont_name)
+        container = storage.get_container(cid)
+        cont_name = pluralize(cont_name)
+
+        response = {
+            'cont_type':    cont_name,
+            '_id':          cid,
+            'label':        container['label'],
+            'parents':      [],
+            'files':        [],
+            'children':     {}
+        }
+
+        if cont_name != 'analysis':
+            analyses = AnalysisStorage().get_analyses(cont_name, cid)
+            response['children']['analyses'] = [{'cont_name': 'analysis', '_id': a['_id'], 'label': a['label']} for a in analyses]
+        else:
+            # Only show output files for analyses
+            container['files'] = [f for f in container.get('files', []) if f.get('output')]
+
+        if cont_name not in ['analyses', 'acquisitions']:
+            children = storage.get_children(cid, projection={'files': 0})
+            response['children'][pluralize(storage.child_cont_name)] = [{'cont_name': storage.child_cont_name, '_id': c['_id'], 'label': c['label']} for c in children]
+
+        parents = storage.get_parent_tree(cid, cont=container)
+        response['parents'] = [{'cont_name': p['cont_type'], '_id': p['_id'], 'label': p.get('label', 'unknown')} for p in parents]
+
+        files = add_suggest_info_to_files(gear, container.get('files', []))
+        response['files'] = [{'name': f['name'], 'suggested': f['suggested']} for f in files]
+
+        return response
```

Comment:
I would prefer this be hidden away in more function calls than logic in the handler, but I don't see any obvious fast way to do that.

Reply:
Yeah it should get broken into it's own method like `build_container_tree` or something and then run `suggest` on the resulting file array. I'll make sure that gets adjusted in the next major change.